### ISSUE TITLE
ModuleNotFoundError: No module named 'PIL'

### ISSUE THREAD

**Issue by aram535** [sgriffin53/ravenopeningsexplorer#2]
Trying to install it in venv under 3.9.2: 

```
Collecting soupsieve==2.2
  Using cached soupsieve-2.2-py3-none-any.whl (33 kB)
Collecting sqlparse==0.3.1
  Using cached sqlparse-0.3.1-py2.py3-none-any.whl (40 kB)
Collecting tk-html-widgets==0.4.0
  Using cached tk_html_widgets-0.4.0.tar.gz (9.9 kB)
    ERROR: Command errored out with exit status 1:
     command: 'f:\github\ravenopeningsexplorer\venv\scripts\python.exe' -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\Aram\\AppData\\Local\\Temp\\pip-install-sahh0pk3\\tk-html-widgets_a942cafd5976490aadb70c083f241780\\setup.py'"'"'; __file__='"'"'C:\\Users\\Aram\\AppData\\Local\\Temp\\pip-install-sahh0pk3\\tk-html-widgets_a942cafd5976490aadb70c083f241780\\setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base 'C:\Users\Aram\AppData\Local\Temp\pip-pip-egg-info-3uopv_3_'
         cwd: C:\Users\Aram\AppData\Local\Temp\pip-install-sahh0pk3\tk-html-widgets_a942cafd5976490aadb70c083f241780\
    Complete output (9 lines):
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "C:\Users\Aram\AppData\Local\Temp\pip-install-sahh0pk3\tk-html-widgets_a942cafd5976490aadb70c083f241780\setup.py", line 2, in <module>
        import tk_html_widgets
      File "C:\Users\Aram\AppData\Local\Temp\pip-install-sahh0pk3\tk-html-widgets_a942cafd5976490aadb70c083f241780\tk_html_widgets\__init__.py", line 8, in <module>
        from tk_html_widgets import html_parser
      File "C:\Users\Aram\AppData\Local\Temp\pip-install-sahh0pk3\tk-html-widgets_a942cafd5976490aadb70c083f241780\tk_html_widgets\html_parser.py", line 9, in <module>
        from PIL import Image, ImageTk
    ModuleNotFoundError: No module named 'PIL'
    ----------------------------------------
WARNING: Discarding https://files.pythonhosted.org/packages/d4/52/ac676bd245a608cf93018d56758f378cf70c475007d07d3707fe476f9fcc/tk_html_widgets-0.4.0.tar.gz#sha256=ddf63feb83619b9a30e3908b97d5a8acacf618391a372d068225e6bf38b1073f (from https://pypi.org/simple/tk-html-widgets/) (requires-python:>=3.4.*). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
ERROR: Could not find a version that satisfies the requirement tk-html-widgets==0.4.0
ERROR: No matching distribution found for tk-html-widgets==0.4.0
```
I don't see any other requirements so if there is some please list them in the README or other doc.

**Comment by sgriffin53** [comment]
Hi,

It seems that you don't have the Pillow module installed. You can install it with 'pip install Pillow'. I've added the requirement to the README and the requirement is already in requirements.txt.

It should work once Pillow is installed.

**Comment by aram535** [comment]
Nope, if you don't want the issues and are just going to close them I'll drop it ... just say so.

```
Requirement already satisfied: soupsieve==2.2 in f:\github\ravenopeningsexplorer\venv\lib\site-packages (from -r requirements.txt (line 24)) (2.2)
Requirement already satisfied: sqlparse==0.3.1 in f:\github\ravenopeningsexplorer\venv\lib\site-packages (from -r requirements.txt (line 25)) (0.3.1)
Collecting tk-html-widgets==0.4.0
  Using cached tk_html_widgets-0.4.0.tar.gz (9.9 kB)
    ERROR: Command errored out with exit status 1:
     command: 'f:\github\ravenopeningsexplorer\venv\scripts\python.exe' -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\Aram\\AppData\\Local\\Temp\\pip-install-z_p1ngk9\\tk-html-widgets_556c6a68f7b94dc09d7bcf35ae24f3da\\setup.py'"'"'; __file__='"'"'C:\\Users\\Aram\\AppData\\Local\\Temp\\pip-install-z_p1ngk9\\tk-html-widgets_556c6a68f7b94dc09d7bcf35ae24f3da\\setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base 'C:\Users\Aram\AppData\Local\Temp\pip-pip-egg-info-63fomoqc'
         cwd: C:\Users\Aram\AppData\Local\Temp\pip-install-z_p1ngk9\tk-html-widgets_556c6a68f7b94dc09d7bcf35ae24f3da\
    Complete output (9 lines):
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "C:\Users\Aram\AppData\Local\Temp\pip-install-z_p1ngk9\tk-html-widgets_556c6a68f7b94dc09d7bcf35ae24f3da\setup.py", line 2, in <module>
        import tk_html_widgets
      File "C:\Users\Aram\AppData\Local\Temp\pip-install-z_p1ngk9\tk-html-widgets_556c6a68f7b94dc09d7bcf35ae24f3da\tk_html_widgets\__init__.py", line 8, in <module>
        from tk_html_widgets import html_parser
      File "C:\Users\Aram\AppData\Local\Temp\pip-install-z_p1ngk9\tk-html-widgets_556c6a68f7b94dc09d7bcf35ae24f3da\tk_html_widgets\html_parser.py", line 9, in <module>
        from PIL import Image, ImageTk
    ModuleNotFoundError: No module named 'PIL'
    ----------------------------------------
WARNING: Discarding https://files.pythonhosted.org/packages/d4/52/ac676bd245a608cf93018d56758f378cf70c475007d07d3707fe476f9fcc/tk_html_widgets-0.4.0.tar.gz#sha256=ddf63feb83619b9a30e3908b97d5a8acacf618391a372d068225e6bf38b1073f (from https://pypi.org/simple/tk-html-widgets/) (requires-python:>=3.4.*). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
ERROR: Could not find a version that satisfies the requirement tk-html-widgets==0.4.0
ERROR: No matching distribution found for tk-html-widgets==0.4.0

```

**Comment by sgriffin53** [comment]
Sorry, I closed it because I was sure installing Pillow would resolve it. I wasn't trying to dismiss your issue.

The "No module named 'PIL'" error is caused because the Pillow module isn't installed. What happens when you run 'pip install Pillow'? Does it say it's already installed?

Sorry if this isn't helpful, but I'm not sure what else could be causing it apart from the missing Pillow module.

**Comment by aram535** [comment]
No problem, new repo, new repo owner .. just wanted to make sure before I waste both our time in case you weren't interested. 

I went heavy handed and replaced all == with >= in the requirements.txt .. that let me install Pillow (v8) and requests, both of which failed for python 3.9 with the hard == requirement.

After that I ran into the new pip depdency-tree issue so I had to wipe my venv and start over with: 
`pip --use-feature=2020-resolver install -r requirements.txt`

That got everything installed.
